Improved AWS and Azure discoverer

oxisto opened the pull request:
This PR improves the discoverers, mainly for AWS and Azure. It introduces discovery of AWS network interfaces and volumes.

oxisto commented:
Blocked by #821

anatheka commented:
@oxisto I have fixed the tests and changed the generation for the storage blob and files client. If this fits, I can do the same for the others in Azure.

anatheka commented:
I have added the missing labels to K8S, except the label for the volume assigned to a pod. The volume label we only get directly from the related storage. The question is, if that is than the right label or if we should use the pod label instead.

oxisto commented:
> @oxisto I have fixed the tests and changed the generation for the storage blob and files client. If this fits, I can do the same for the others in Azure.

Ah good! The tests still seem to fail because I moved some of the parts related to Azure Disks from storage to compute and now the tests cannot find the necessary mocks.

anatheka commented:
> @oxisto I have fixed the tests and changed the generation for the storage blob and files client. If this fits, I can do the same for the others in Azure.

Should I refactor the Azure client generation for compute and network as well?

oxisto commented on service/discovery/aws/compute.go:
```diff
@@ -253,11 +364,45 @@ func (*computeDiscovery) getNameOfVM(vm *typesEC2.Instance) string {
+	return aws.ToString(vm.VolumeId)
+}
+
+// getNameOfVolume returns the name if exists (i.e. a tag with key 'name' exists), otherwise instance ID is used
```
wrong documentation text

oxisto commented on service/discovery/aws/compute.go:
```diff
@@ -253,11 +364,45 @@ func (*computeDiscovery) getNameOfVM(vm *typesEC2.Instance) string {
-// addARNToVM generates the ARN of a VM instance
-func (d computeDiscovery) addARNToVM(vm *typesEC2.Instance) voc.ResourceID {
+// getNameOfVolume returns the name if exists (i.e. a tag with key 'name' exists), otherwise instance ID is used
+func (*computeDiscovery) getNameOfVolume(vm *typesEC2.Volume) string {
```
this is probably absolute now that we have `nameOrId`

oxisto commented on service/discovery/azure/azure.go:
```diff
@@ -82,6 +85,12 @@ type azureDiscovery struct {
+	client              client
+}
+
+type client struct {
```
I would suggest `clients`

oxisto commented on service/discovery/azure/azure.go:
```diff
@@ -82,6 +85,12 @@ type azureDiscovery struct {
 	cred                azcore.TokenCredential
 	clientOptions       arm.ClientOptions
 	discovererComponent string
+	client              client
```
`clients`

oxisto commented on service/discovery/azure/storage.go:
```diff
@@ -523,3 +378,35 @@ func accountName(id string) string {
 }
+
+// initializeBlobContainerClient creates the client if not already exists
+func (d *azureStorageDiscovery) initializeBlobContainerClient() (err error) {
```
I think we just used `initXXX` somewhere else

anatheka replied:
You mean the naming or do you wanted to use the other generic function? 

Is it possible to change these methods to generic methods?

oxisto replied:
I was referring to the naming. Would be interesting if these could be generic functions.

oxisto commented on service/discovery/azure/storage.go:
```diff
@@ -523,3 +378,35 @@ func accountName(id string) string {
+}
+
+// initializeFileStorageClient creates the client if not already exists
+func (d *azureStorageDiscovery) initializeFileStorageClient() (err error) {
```
same as above

oxisto commented on service/discovery/discovery.go:
```diff
@@ -69,6 +69,16 @@ type grpcTarget struct {
+const DiscoveryEventTypeDiscovererStart = 0
+const DiscoveryEventTypeDiscovererFinished = 1
+
+type DiscoveryEvent struct {
```
missing documentation

oxisto reviewed (commented):
Some review comments, some of which I should probably address myself :)

anatheka reviewed (approved):
:+1:
